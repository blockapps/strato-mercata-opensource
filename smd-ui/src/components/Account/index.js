import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import HexText from '../HexText';

class Account extends Component {
  render() {
    const {
      name,
      address,
      account
    } = this.props;

    return (
      <div className="pt-card address-margin-bottom" key={address}>
        <div className="row smd-pad-2 smd-margin-4 smd-vertical-center">
          <div className="col-sm-10">
            <h4>
              Address: &nbsp;&nbsp; <HexText value={address} classes="smd-pad-2" />
            </h4>
          </div>
        </div>

        <table className="pt-table pt-str">
          <thead>
            <tr>
              <th>Field</th>
              <th>Value</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td><strong>Contract Root</strong></td>
              <td><HexText value={account.contractRoot} classes="smd-pad-2" /></td>
            </tr>
            <tr>
              <td><strong>Kind</strong></td>
              <td>{account.kind}</td>
            </tr>
            <tr>
              <td><strong>Balance</strong></td>
              <td>{account.balance && `${account.balance} wei`}</td>
            </tr>
            <tr>
              <td><strong>Latest Block Number</strong></td>
              <td>{account.latestBlockNum}</td>
            </tr>
            <tr>
              <td><strong>Code Hash</strong></td>
              <td><HexText value={account.codeHash} classes="smd-pad-2" /></td>
            </tr>
            <tr>
              <td><strong>Nonce</strong></td>
              <td>{account.nonce}</td>
            </tr>
          </tbody>
        </table>
      </div>
    );
  }
}

export function mapStateToProps(state, ownProps) {
  const name = ownProps.name;
  const address = ownProps.address;
  const accounts = state.accounts.accounts;
  return {
    account: Object.getOwnPropertyNames(accounts).indexOf(name) >= 0 ? state.accounts.accounts[name][address] : {},
  };
}

export default withRouter(
  connect(
    mapStateToProps,
  )(Account)
);
