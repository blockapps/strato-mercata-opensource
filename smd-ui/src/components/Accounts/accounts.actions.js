export const FETCH_ACCOUNTS = 'FETCH_ACCOUNTS';
export const FETCH_ACCOUNTS_SUCCESSFULL = 'FETCH_ACCOUNTS_SUCCESSFULL';
export const FETCH_ACCOUNTS_FAILED = 'FETCH_ACCOUNTS_FAILED';
export const CHANGE_ACCOUNT_FILTER = 'CHANGE_ACCOUNT_FILTER';
export const FETCH_ACCOUNT_ADDRESS_REQUEST = 'FETCH_ACCOUNT_ADDRESS_REQUEST';
export const FETCH_USER_ADDRESSES_SUCCESSFUL = 'FETCH_ACCOUNT_ADDRESS_SUCCESS';
export const FETCH_USER_ADDRESSES_FAILED = 'FETCH_ACCOUNT_ADDRESS_FAILURE';
export const FETCH_ACCOUNT_DETAIL_REQUEST = 'FETCH_ACCOUNT_DETAIL_REQUEST';
export const FETCH_ACCOUNT_DETAIL_SUCCESS = 'FETCH_ACCOUNT_DETAIL_SUCCESS';
export const FETCH_ACCOUNT_DETAIL_FAILURE = 'FETCH_ACCOUNT_DETAIL_FAILURE';
export const RESET_ACCOUNT_ADDRESS = 'RESET_ACCOUNT_ADDRESS';
export const GET_BALANCE = 'GET_BALANCE';
export const BALANCE_SUCCESS = 'BALANCE_SUCCESS';
export const BALANCE_FAILURE = 'BALANCE_FAILURE';
export const FETCH_CURRENT_ACCOUNT_DETAIL_REQUEST = 'FETCH_CURRENT_ACCOUNT_DETAIL_REQUEST';
export const FETCH_CURRENT_ACCOUNT_DETAIL_SUCCESS = 'FETCH_CURRENT_ACCOUNT_DETAIL_SUCCESS';
export const FETCH_CURRENT_ACCOUNT_DETAIL_FAILURE = 'FETCH_CURRENT_ACCOUNT_DETAIL_FAILURE';
export const FETCH_OAUTH_ACCOUNTS_REQUEST = 'FETCH_OAUTH_ACCOUNTS_REQUEST';
export const FETCH_OAUTH_ACCOUNTS_SUCCESS = 'FETCH_OAUTH_ACCOUNTS_SUCCESS';
export const FETCH_OAUTH_ACCOUNTS_FAILURE = 'FETCH_OAUTH_ACCOUNTS_FAILURE';

export const fetchAccounts = function (loadAddresses, loadBalances, chainId) {
  return {
    type: FETCH_ACCOUNTS,
    loadAddresses: loadAddresses,
    loadBalances: loadBalances,
    chainId: chainId
  }
};

export const fetchAccountsSuccess = function (accounts) {
  return {
    type: FETCH_ACCOUNTS_SUCCESSFULL,
    accounts: accounts
  }
};

export const fetchAccountsFailure = function (error) {
  return {
    type: FETCH_ACCOUNTS_FAILED,
    error: error,
  }
};

export const changeAccountFilter = function (filter) {
  return {
    type: CHANGE_ACCOUNT_FILTER,
    filter: filter
  }
};

export const fetchUserAddresses = function (name, loadBalances, chainId) {
  return {
    type: FETCH_ACCOUNT_ADDRESS_REQUEST,
    name: name,
    loadBalances: loadBalances,
    chainId: chainId
  }
};

export const resetUserAddress = function (name) {
  return {
    type: RESET_ACCOUNT_ADDRESS,
    name: name
  }
}

export const fetchUserAddressesSuccess = function (name, addresses) {
  return {
    type: FETCH_USER_ADDRESSES_SUCCESSFUL,
    name: name,
    addresses: addresses
  }
};

export const fetchUserAddressesFailure = function (name, error) {
  return {
    type: FETCH_USER_ADDRESSES_FAILED,
    name: name,
    error: error
  }
};

export const fetchAccountDetail = function (name, address, chainId, flag) {
  return {
    type: FETCH_ACCOUNT_DETAIL_REQUEST,
    name: name,
    address: address,
    chainId: chainId,
    flag: flag
  }
};

export const fetchAccountDetailSuccess = function (name, address, detail) {
  return {
    type: FETCH_ACCOUNT_DETAIL_SUCCESS,
    name: name,
    address: address,
    detail: detail
  }
};

export const fetchAccountDetailFailure = function (name, address, error) {
  return {
    type: FETCH_ACCOUNT_DETAIL_FAILURE,
    name: name,
    address: address,
    error: error
  }
};

export const fetchBalanceRequest = function (address) {
  return {
    type: GET_BALANCE,
    address
  }
}

export const fetchBalanceSuccess = function (detail) {
  return {
    type: BALANCE_SUCCESS,
    detail
  }
};

export const fetchBalanceFailure = function (error) {
  return {
    type: BALANCE_FAILURE,
    error
  }
};

export const fetchCurrentAccountDetail = function (address) {
  return {
    type: FETCH_CURRENT_ACCOUNT_DETAIL_REQUEST,
    address: address
  }
};

export const fetchCurrentAccountDetailSuccess = function (address, detail) {
  return {
    type: FETCH_CURRENT_ACCOUNT_DETAIL_SUCCESS,
    address: address,
    detail: detail
  }
};

export const fetchCurrentAccountDetailFailure = function (address, error) {
  return {
    type: FETCH_CURRENT_ACCOUNT_DETAIL_FAILURE,
    address: address,
    error: error
  }
};

export const fetchOauthAccounts = function () {
  return {
    type: FETCH_OAUTH_ACCOUNTS_REQUEST
  }
};

export const fetchOauthAccountsSuccess = function (data) {
  return {
    type: FETCH_OAUTH_ACCOUNTS_SUCCESS,
    data
  }
};

export const fetchOauthAccountsFailure = function () {
  return {
    type: FETCH_OAUTH_ACCOUNTS_FAILURE
  }
};
