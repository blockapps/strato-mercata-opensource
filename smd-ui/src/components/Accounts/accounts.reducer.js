import {
  FETCH_ACCOUNTS,
  FETCH_ACCOUNTS_SUCCESSFULL,
  FETCH_ACCOUNTS_FAILED,
  CHANGE_ACCOUNT_FILTER,
  FETCH_USER_ADDRESSES_SUCCESSFUL,
  FETCH_USER_ADDRESSES_FAILED,
  FETCH_ACCOUNT_DETAIL_SUCCESS,
  FETCH_ACCOUNT_DETAIL_FAILURE,
  RESET_ACCOUNT_ADDRESS,
  BALANCE_SUCCESS,
  BALANCE_FAILURE,
  FETCH_CURRENT_ACCOUNT_DETAIL_SUCCESS,
  FETCH_CURRENT_ACCOUNT_DETAIL_FAILURE,
  FETCH_OAUTH_ACCOUNTS_SUCCESS,
  FETCH_OAUTH_ACCOUNTS_FAILURE
} from './accounts.actions';

const initialState = {
  accounts: {},
  currentAccountDetail: null,
  filter: '',
  error: null,
  oauthAccounts: []
};

const reducer = function (state = initialState, action) {
  switch (action.type) {
    case FETCH_ACCOUNTS:
      return {
        ...state,
        accounts: state.accounts,
        filter: state.filter,
        error: null,
        currentUserBalance: state.currentUserBalance
      };
    case FETCH_ACCOUNTS_SUCCESSFULL:
      const accounts = action.accounts.reduce(function (result, item) {
        result[item] = {};
        return result;
      }, {});
      return {
        ...state,
        accounts: accounts,
        filter: state.filter,
        error: null,
      };
    case FETCH_ACCOUNTS_FAILED:
      return {
        ...state,
        accounts: state.accounts,
        filter: state.filter,
        error: action.error
      };
    case CHANGE_ACCOUNT_FILTER:
      return {
        ...state,
        accounts: state.accounts,
        filter: action.filter,
        error: state.error,
      }
    case FETCH_USER_ADDRESSES_SUCCESSFUL:
      const addresses = action.addresses.reduce(function (result, address) {
        result[address] = {
          error: null
        }
        return result;
      }, {})
      return {
        ...state,
        accounts: {
          ...state.accounts,
          [action.name]: addresses
        },
        filter: state.filter,
        error: state.error,
        currentUserBalance: state.currentUserBalance
      }
    case FETCH_USER_ADDRESSES_FAILED:
      return {
        ...state,
        accounts: {
          ...state.accounts,
          [action.name]: {
            error: action.error
          }
        },
        filter: state.filter,
        error: state.error,
        currentUserBalance: state.currentUserBalance
      }
    case FETCH_ACCOUNT_DETAIL_SUCCESS:
      return {
        ...state,
        accounts: {
          ...state.accounts,
          [action.name]: {
            ...state.accounts[action.name],
            [action.address]: {
              ...action.detail,
              error: null
            }
          }
        },
        filter: state.filter,
        error: state.error,
        currentUserBalance: state.currentUserBalance
      }
    case RESET_ACCOUNT_ADDRESS:
      return {
        ...state,
        accounts: {
          ...state.accounts,
          [action.name]: {}
        },
        filter: state.filter,
        error: state.error
      }
    case FETCH_ACCOUNT_DETAIL_FAILURE:
      return {
        ...state,
        accounts: {
          ...state.accounts,
          [action.name]: {
            ...state.accounts[action.name],
            [action.address]: {
              error: action.error
            }
          }
        },
        filter: state.filter,
        error: state.error,
        currentUserBalance: state.currentUserBalance
      }
    case BALANCE_SUCCESS:
      return {
        ...state,
        currentUserBalance: action.detail && action.detail.balance
      }
    case BALANCE_FAILURE:
      return {
        ...state,
        currentUserBalance: null
      }
    case FETCH_CURRENT_ACCOUNT_DETAIL_SUCCESS:
      return {
        ...state,
        currentAccountDetail: action.detail,
        error: null
      }
    case FETCH_CURRENT_ACCOUNT_DETAIL_FAILURE:
      return {
        ...state,
        currentAccountDetail: null,
        error: action.error
      }
    case FETCH_OAUTH_ACCOUNTS_SUCCESS:
      return {
        ...state,
        oauthAccounts: action.data
      }
    case FETCH_OAUTH_ACCOUNTS_FAILURE:
      return {
        ...state,
        oauthAccounts: [],
        error: action.error
      }
    default:
      return state;
  }
};

export default reducer;
