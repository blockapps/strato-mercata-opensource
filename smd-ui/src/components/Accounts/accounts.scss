.accounts-margin-top {
  bottom: 10px;
  padding-right: 7px;
  padding-left: 10px;
  overflow-y: auto;
  top: 170px;
}

.main-div {
  overflow-y: auto;
  margin-top: 40px;
  height: 70vh;
}

.main-div .smd-margin-8:first-of-type {
  margin-top: 0px;
}

.main-div::-webkit-scrollbar {
  width: 10px;
}

.account-details {
  margin-top: 40px;
  overflow-y: auto;
  height: 70vh;
  overflow-x: hidden;
}

.account-details::-webkit-scrollbar {
  width: 10px;
}

::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 6px rgba(0,0,0,0.3);
  border-radius: 4px;
  background: #2F424D; 
}
 
::-webkit-scrollbar-thumb {
  border-radius: 4px; 
  background: #F4F8FA; 
}

::-webkit-scrollbar-thumb:hover {
  background: #BBCBD4; 
}

.send-tokens-dialog .chain-field {
  width: 100%;
}