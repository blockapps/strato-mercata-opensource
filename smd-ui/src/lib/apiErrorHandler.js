export function handleApiError(response) {
  return new Promise(function(resolve,reject){
    if(response.error) {
      reject(response.error);
    }
    else {
      resolve(response);
    }
  })
}
