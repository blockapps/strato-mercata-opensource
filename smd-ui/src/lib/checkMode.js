import { env } from "../env";

export function isOauthEnabled() {
  return env.OAUTH_ENABLED
}