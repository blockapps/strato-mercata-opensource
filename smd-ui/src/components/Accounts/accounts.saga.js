import {
  takeLatest,
  takeEvery,
  put,
  call,
  cancelled
} from 'redux-saga/effects';
import {
  FETCH_ACCOUNTS,
  FETCH_ACCOUNT_ADDRESS_REQUEST,
  FETCH_ACCOUNT_DETAIL_REQUEST,
  GET_BALANCE,
  fetchAccountsSuccess,
  fetchAccountsFailure,
  fetchUserAddresses,
  fetchUserAddressesSuccess,
  fetchUserAddressesFailure,
  fetchAccountDetail,
  fetchAccountDetailSuccess,
  fetchAccountDetailFailure,
  fetchBalanceSuccess,
  fetchBalanceFailure,
  fetchCurrentAccountDetailSuccess,
  fetchCurrentAccountDetailFailure,
  FETCH_CURRENT_ACCOUNT_DETAIL_REQUEST,
  FETCH_OAUTH_ACCOUNTS_REQUEST,
  fetchOauthAccountsSuccess,
  fetchOauthAccountsFailure
} from './accounts.actions';
import { env } from '../../env';
import { hideLoading } from 'react-redux-loading-bar';
import { handleErrors } from '../../lib/handleErrors';
import { createUrl } from '../../lib/url';

const oauthAccountDataUrl = env.STRATO_URL_V23 + "/users";
const accountDataUrl = env.STRATO_URL + "/account";
const usernameUrl = env.BLOC_URL + "/users";

export function getAccountsApi() {
  return fetch(
    usernameUrl,
    {
      method: 'GET',
      credentials: "include",
      headers: {
        'Accept': 'application/json'
      },
    })
    .then(handleErrors)
    .then(function (response) {
      return response.json()
    })
    .catch(function (error) {
      throw error;
    })
}

export function getUserAddressesApi(username) {
  const options = { params: { user: username } };
  const url = env.BLOC_URL + createUrl("/users/::user", options);

  return fetch(
    url,
    {
      method: 'GET',
      credentials: "include",
      headers: {
        'Accept': 'application/json'
      },
    }
  )
    .then(handleErrors)
    .then(function (response) {
      return response.json();
    })
    .catch(function (error) {
      throw error;
    });
}

export function getAccountDetailApi(address, chainid) {
  const options = { query: { address, chainid } };
  const url = createUrl(accountDataUrl, options);

  return fetch(
    url,
    {
      method: 'GET',
      credentials: "include",
      headers: {
        'Accept': 'application/json'
      },
    }
  )
    .then(handleErrors)
    .then(function (response) {
      return response.json();
    })
    .catch(function (error) {
      throw error;
    });
}

export function getOauthAccountsApi() {
  const cirrusUrl = env.CIRRUS_URL + "/Certificate?userAddress=eq.";


  // strato URL add limit and offset if needed
  return fetch(
    oauthAccountDataUrl,
    {
      method: 'GET',
      credentials: "include",
      headers: {
        'Accept': 'application/json'
      },
    }
  )
    .then(handleErrors)
    .then(function (res) {
      return res.json();
    })
    .then(function (users) {
      const fetches = [];
      for (let x in users) {
        const url = cirrusUrl + users[x].address;
        fetches.push(
          fetch (
            url,
            {
              method: 'GET',
              credentials: "include",
              headers: {
                'Accept': 'application/json'
              },
            }
          )
            .then(handleErrors)
            .then(function (response) {
              return response.json();
            })
            .catch(function (error) {
              throw error;
            }))
      }
      return Promise.all(fetches).then(function (responses) {
        return responses.map(x => x.length > 0 ? x[0] : null);
      });
    })
    .catch(function (error) {
      throw error;
    });
}

export function* getAccounts(action) {
  try {
    const response = yield call(getAccountsApi);
    // const response = ["sz1152", "sz2699"];
    yield put(fetchAccountsSuccess(response));
    // dispatch the action if necessary
    if (action.loadAddresses && response.length > 0) {
      yield put(fetchUserAddresses(response[0], action.loadBalances, action.chainId));
    }
  }
  catch (err) {
    yield put(fetchAccountsFailure(err));
  } finally {
    if (yield cancelled()) {
      yield put(hideLoading());
    }
  }
}

export function* getUserAddresses(action) {
  try {
    const response = yield call(getUserAddressesApi, action.name);
    // const response = ["999"];
    yield put(fetchUserAddressesSuccess(action.name, response));
    if (action.loadBalances) {
      yield response.map(address => put(fetchAccountDetail(action.name, address, action.chainId)));
    }
  }
  catch (err) {
    yield put(fetchUserAddressesFailure(action.name, err));
  }
}

export function* getAccountDetail(action) {
  try {
    const response = yield call(getAccountDetailApi, action.address, action.chainId);
    // don't ask about response['0'].
    yield put(fetchAccountDetailSuccess(action.name, action.address, response['0']));
  }
  catch (err) {
    yield put(fetchAccountDetailFailure(action.name, action.address, err));
  }
}

export function* getCurrentAccountDetail(action) {
  try {
    const response = yield call(getAccountDetailApi, action.address);
    // don't ask about response['0'].
    yield put(fetchCurrentAccountDetailSuccess(action.address, response['0']));
  }
  catch (err) {
    yield put(fetchCurrentAccountDetailFailure(action.address, err));
  }
}

export function* getBalance(action) {
  try {
    const response = yield call(getAccountDetailApi, action.address);
    yield put(fetchBalanceSuccess(response['0']));
  }
  catch (err) {
    yield put(fetchBalanceFailure(err));
  }
}

export function* getOauthAccounts() {
  try {
    const response = yield call(getOauthAccountsApi);
    yield put(fetchOauthAccountsSuccess(response));
  }
  catch (err) {
    yield put(fetchOauthAccountsFailure('failed to fetch oauth accounts'));
  }
}

export default function* watcAccountActions() {
  yield [
    takeLatest(FETCH_ACCOUNTS, getAccounts),
    takeEvery(FETCH_ACCOUNT_ADDRESS_REQUEST, getUserAddresses),
    takeEvery(FETCH_ACCOUNT_DETAIL_REQUEST, getAccountDetail),
    takeEvery(FETCH_CURRENT_ACCOUNT_DETAIL_REQUEST, getCurrentAccountDetail),
    takeEvery(GET_BALANCE, getBalance),
    takeEvery(FETCH_OAUTH_ACCOUNTS_REQUEST, getOauthAccounts)
  ];
}
