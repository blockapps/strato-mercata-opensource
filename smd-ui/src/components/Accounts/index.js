import React, { Component } from 'react';
import {
  fetchAccounts,
  fetchOauthAccounts,
} from './accounts.actions';
import mixpanelWrapper from '../../lib/mixpanelWrapper';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Tour from '../Tour';
import BlocAccounts from './components/BlocAccounts';
import OauthAccounts from './components/OauthAccounts';
import './accounts.css';
import { isOauthEnabled } from '../../lib/checkMode';
import ReactGA from 'react-ga4';

const tourSteps = [/* {
    title: 'Create User',
    text: 'Create a user here',
    selector: '#accounts-create-user-button',
    position: 'bottom', type: 'hover',
    isFixed: true,
  }, ) */
  {
    title: 'Upload a Smart Contract',
    text: 'Drag and drop a <strong>.sol</strong> file, and you will be able to manage your ' +
      'Smart Contract from within the STRATO dashboard.',
    /* text: '<div class="inline-code-sample">contract RentSplit {<br>address <strong>Roommate 1</strong>;<br><strong>Roommate 2</strong>;<br><strong>Roommate 3</strong>;<br>mapping (address => uint) RentSplit;<br></div>', */
    selector: '#contracts',
    position: 'bottom',
    isFixed: true
  }
];

class Accounts extends Component {

  constructor() {
    super()
    this.state = {
      selected: 0
    }
  }

  componentDidMount() {
    if (isOauthEnabled()) {
      this.props.fetchOauthAccounts();
    } else {
      this.props.fetchAccounts(true, true, this.props.selectedChain);
    }
    mixpanelWrapper.track('accounts_page_load');
    ReactGA.send({hitType: "pageview", page: "/accounts", title: "Accounts"});
  }

  render() {
    return (
      <div className="container-fluid pt-dark">
        <Tour
          name="accounts"
          steps={tourSteps}
          finalStepSelector='#contracts'
          nextPage='chains' />
        {isOauthEnabled() ? <OauthAccounts /> : <BlocAccounts />}
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {};
}

export default withRouter(
  connect(mapStateToProps,
    {
      fetchOauthAccounts,
      fetchAccounts
    }
  )(Accounts));
