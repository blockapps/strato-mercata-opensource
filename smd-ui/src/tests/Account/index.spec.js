import React from 'react';
import Account, { mapStateToProps } from '../../components/Account/index';
import { indexAccountsMock } from '../Accounts/accountsMock';
import { accountDetails } from './accountMock';
import * as checkMode from '../../lib/checkMode';

describe('Account: index', () => {

  describe('renders correctly', () => {

    beforeAll(() => {
      checkMode.isOauthEnabled = jest.fn().mockReturnValue(false);
    })

    test('with empty values', () => {
      const props = {
        account: accountDetails,
        name: '',
        address: '',
      };

      const wrapper = shallow(
        <Account.WrappedComponent {...props} />
      );
      expect(wrapper.debug()).toMatchSnapshot();
    });

    test('with values', () => {
      const props = {
        account: accountDetails,
        name: '',
        address: '0004537908d44f458acb24b0f2c863ccd2bd3a13',
      };

      const wrapper = shallow(
        <Account.WrappedComponent {...props} />
      );
      expect(wrapper.debug()).toMatchSnapshot();

      // Testing that the leading zeroes are not truncated
      expect(wrapper.find('HexText').first().html()).toContain('<div class="pt-text-overflow-ellipsis">' + props.address)
    })
  })

  describe('mapStateToProps', () => {
    test('with values', () => {
      const state = {
        accounts: {
          accounts: indexAccountsMock,
        }
      }
      const ownProps = {
        name: 'Supplier1',
        address: '370adf114257cb0e0025eedf0a96261b51af23e3'
      }
      expect(mapStateToProps(state, ownProps)).toMatchSnapshot();
    });

    test('without values', () => {
      const state = {
        accounts: {
          accounts: indexAccountsMock,
        }
      }
      const ownProps = {
        name: 'Supply',
        address: '370adf114257cb0e0025eedf0a96261b51af23e3'
      }
      expect(mapStateToProps(state, ownProps)).toMatchSnapshot();
    });
  })

});