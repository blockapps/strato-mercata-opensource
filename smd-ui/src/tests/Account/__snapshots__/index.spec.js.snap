// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account: index mapStateToProps with values 1`] = `
Object {
  "account": Object {
    "error": null,
  },
}
`;

exports[`Account: index mapStateToProps without values 1`] = `
Object {
  "account": Object {},
}
`;

exports[`Account: index renders correctly with empty values 1`] = `
"<div className=\\"pt-card address-margin-bottom\\">
  <div className=\\"row smd-pad-2 smd-margin-4 smd-vertical-center\\">
    <div className=\\"col-sm-10\\">
      <h4>
        Address:    
        <HexText value=\\"\\" classes=\\"smd-pad-2\\" />
      </h4>
    </div>
  </div>
  <table className=\\"pt-table pt-str\\">
    <thead>
      <tr>
        <th>
          Field
        </th>
        <th>
          Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <strong>
            Contract Root
          </strong>
        </td>
        <td>
          <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
        </td>
      </tr>
      <tr>
        <td>
          <strong>
            Kind
          </strong>
        </td>
        <td />
      </tr>
      <tr>
        <td>
          <strong>
            Balance
          </strong>
        </td>
        <td />
      </tr>
      <tr>
        <td>
          <strong>
            Latest Block Number
          </strong>
        </td>
        <td />
      </tr>
      <tr>
        <td>
          <strong>
            Code Hash
          </strong>
        </td>
        <td>
          <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
        </td>
      </tr>
      <tr>
        <td>
          <strong>
            Nonce
          </strong>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`Account: index renders correctly with values 1`] = `
"<div className=\\"pt-card address-margin-bottom\\">
  <div className=\\"row smd-pad-2 smd-margin-4 smd-vertical-center\\">
    <div className=\\"col-sm-10\\">
      <h4>
        Address:    
        <HexText value=\\"0004537908d44f458acb24b0f2c863ccd2bd3a13\\" classes=\\"smd-pad-2\\" />
      </h4>
    </div>
  </div>
  <table className=\\"pt-table pt-str\\">
    <thead>
      <tr>
        <th>
          Field
        </th>
        <th>
          Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <strong>
            Contract Root
          </strong>
        </td>
        <td>
          <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
        </td>
      </tr>
      <tr>
        <td>
          <strong>
            Kind
          </strong>
        </td>
        <td />
      </tr>
      <tr>
        <td>
          <strong>
            Balance
          </strong>
        </td>
        <td />
      </tr>
      <tr>
        <td>
          <strong>
            Latest Block Number
          </strong>
        </td>
        <td />
      </tr>
      <tr>
        <td>
          <strong>
            Code Hash
          </strong>
        </td>
        <td>
          <HexText value={[undefined]} classes=\\"smd-pad-2\\" />
        </td>
      </tr>
      <tr>
        <td>
          <strong>
            Nonce
          </strong>
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</div>"
`;
